### ISSUE TITLE
CMSIS RTOS: Documentation about preemptive scheduling

### ISSUE THREAD

**Issue by senselessDev** [ARM-software/CMSIS#98]
I was wondering why there is no word of what is happening when `#define OS_ROBIN` is set to 0 in [this](http://www.keil.com/pack/doc/CMSIS/RTX/html/_system_config.html) documentation. I guess then the normal preemptive scheduling is active.

**Comment by senselessDev** [comment]
Unfortunatley, it is not working as i expected. If a higher priority thread is waiting like that: `evt = osMessageGet(msgQid_control_error, osWaitForever);` I thought lower priority threads should be scheduled if there is nothing in the queue. Am I wrong?

**Comment by senselessDev** [comment]
> If a higher priority thread is waiting like that: evt = osMessageGet(msgQid_control_error, osWaitForever); I thought lower priority threads should be scheduled if there is nothing in the queue and RR is not used. Am I wrong?
Ok, found the problem for this one: missed the osMessageCreate. With usage of this function, the higher priority threads waiting for a message are preempted as expected by lower priority threads. 

TL;DR
But I haven't figured out the exact difference between `#define OS_ROBIN 0` and `#define OS_ROBIN 1` yet, so there should be better documentation about that (so leaving the issue open).
